doc/pfc_asym/PFC_Asymmetric_HLD.md

```diff
@@ -0,0 +1,391 @@
+   * [Python modules](#python-modules)
+   * [Global variables](#global-variables)
+   * [PTF test case execution](#ptf-test-case-execution)
+   * [Pytest fixtures](#pytest-fixtures)
+		* [deploy_pfc_gen](#deploy_pfc_gen)
+		* [setup](#setup)
+		* [pfc_storm_template](#pfc_storm_template)
+		* [pfc_storm_runner](#pfc_storm_runner)
+		* [enable_pfc_asym](#enable_pfc_asym)
+* [Test](#Test)
+* [Test cases](#Test-cases)
+
+## Overview
+The purpose is to test functionality of Asymmetric PFC on the SONIC based DUT, closely resembling production environment.
+
+### Scope
+The test is targeting a running SONIC system with fully functioning configuration. The purpose of the test is to perform functional testing of Asymmetric PFC on SONIC system. There will be reused existed PTF test suite for PFC Asymmetric which is located at ```ansible\roles\test\files\saitests\pfc_asym.py```.
+
+### Testbed
+The test will run on the following testbeds:
+* T0
+
+## Setup configuration
+No setup pre-configuration is required, test will configure and clean-up all the configuration.
+
+## Existed modules refactoring
+There is already existed PFC asymmetric test cases which use PTF to send traffic.
+Test suite location is ```ansible\roles\test\files\saitests\pfc_asym.py```
+
+It requires several updates:
+
+1. Packets sending speed can be increased be using multiprocessing instead of multithreading library.
+
+```PfcAsymBaseTest.send```
+
+Use ```multiprocessing.Process``` instead of ```threading.Thread```.
+
+2. Looks like there is a redundancy in generating configuration file for ARP responder.
+
+```PfcAsymBaseTest.setUpArpResponder```
+
+Fix the loop to store file only once
+
+3. Need to remove dst port drop counters verification.
+
+```PfcAsymOffOnTxTest.runTest```
+
+Remove the following step which is defined by inline comment:
+```Verify that some packets are dropped on dst port, which means that Tx buffer is full```
+
+Because on some platforms packets does not count on HLL while shaper is closed and count on the HLL & SLL as soon as shaper released (but not consistently).
+
+
+## Python  modules to setup and run test
+
+### Python modules
+*New test suite will be developed:*
+
+```tests/pfc_asym/pfc_asym.py```
+
+### Global variables
+```
+PFC_GEN_FILE = "pfc_gen.py"
+PFC_FRAMES_NUMBER = 1000000
+PFC_QUEUE_INDEX = 0xff
+```
+
+### PTF test case execution
+To run existed PTF test cases there will be used ```tests/ptf_runner.py``` module
+
+### Pytest fixtures
+Preparation before test cases run will be executed in the following pytest fixtures:
+```setup, pfc_storm_template, pfc_storm_runner, deploy_pfc_gen```
+
+Description each of the fixture is below.
+
+### deploy_pfc_gen (scope="module", autouse=True)
+
+Deploy ```roles/test/files/helpers/pfc_gen.py``` to the fanout host.
+This step can be different for different platforms. Below there is description how it works for Mellanox and Arista cases, also how to add support of another platform type.
+
```

Comment:
Please also mention that you are not changing logic here and just keep it as is

Reply:
Fixed